tools/ping_enumerator.py

```diff
@@ -136,8 +136,11 @@ def list_serial_devices(self):
 
     # Look at each serial device, probe for ping
     for dev in enumerator.list_serial_devices():
-        link = enumerator.detect_device(dev)
-        if link:
-            enumerator.make_symlink(dev, link)
-        else:
-            print("Unable to identify device at ", dev)
+        try:
+            link = enumerator.detect_device(dev)
+            if link:
+                enumerator.make_symlink(dev, link)
+            else:
+                print("Unable to identify device at ", dev)
+        except Exception as exception:
```

Comment:
What other exception you got here other than the `subprocess.CalledProcessError` that are caught inside detect_device() and make_symlink() ?

Reply:
I don't know how to find this magic names from the backtrace
```
Traceback (most recent call last):                                                                                                                                                    
  File "./ping_enumerator.py", line 139, in <module>                                                                                                                                  
    link = enumerator.detect_device(dev)                                                                                                                                              
  File "./ping_enumerator.py", line 39, in detect_device                                                                                                                              
    ping = PingDevice("/dev/serial/by-id/" + dev, 115200)                                                                                                                             
  File "/usr/local/lib/python2.7/dist-packages/brping/device.py", line 34, in __init__                                                                                                
    raise Exception("Failed to open the given serial port: {0}".format(exception))                                                                                                    
Exception: Failed to open the given serial port: (32, 'Broken pipe') 
```

Reply:
Hah, it is a generic exception, but I looked at the wrong function anyway, disregard that :man_facepalming: .

Anyway, I would rather have `detect_device()` capture the exception and return `None` as it was unable to detect a device at the port

Reply:
Well, that's not related to this PR since is a ping-python functionality, I'm just fixing what is happening with the actual version that we have. 
New issue open:
https://github.com/bluerobotics/ping-python/issues/79

Reply:
I meant capturing it here: https://github.com/bluerobotics/companion/blob/master/tools/ping_enumerator.py#L39

Reply:
Indeed, thanks for the explanation, done.